functional_tests/basic_java_test/tests.py

```diff
@@ -70,19 +70,15 @@ def setUp(self): # pylint: disable=arguments-differ
 
 	def testCompileSucceeds(self):
 		"""Test that the project succesfully compiles"""
-		javaExe = "java{}".format(".exe" if platform.system() == "Windows" else "")
+		self.assertIn("JAVA_HOME", os.environ, "JAVA_HOME must be defined in the environment if the Java binaries are not available from the system path")
+		javaExe = os.path.join(os.environ["JAVA_HOME"], "bin", "java{}".format(".exe" if platform.system() == "Windows" else ""))
 
 		# Verify the Java executable can be found since it's required for running the JAR output for this test.
-		try:
-			subprocess.call([javaExe], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-		except:
-			self.assertIn("JAVA_HOME", os.environ, "JAVA_HOME must be defined in the environment if the Java binaries are not available from the system path")
-			javaExe = os.path.join(os.environ["JAVA_HOME"], "bin", javaExe)
-			self.assertFileExists(javaExe)
-
+		self.assertFileIsExecutable(javaExe)
 		self.assertMakeSucceeds("-v", "--project=hello_world", "--show-commands", "--toolchain=oracle-java")
 
 		self.assertTrue(os.access(self.outputFile, os.F_OK))
-		out = subprocess.check_output([javaExe, "-jar", self.outputFile])
+		ret, out, _ = Run([javaExe, "-jar", self.outputFile])
```

Comment:
I know I'm silly... but I prefer importing modules and calling the function on the module rather than importing functions. It makes it more clear in a review that the function is not local to this file. I make exceptions for things like PlatformString and PlatformBytes because they're mimicking built-in str and bytes functions, but most things, I prefer to have the module name in the call site.